[Bug]: Missing file pyplot.pyi for mypy typing

### Bug summary

With `mypy`, `plt.show()` results in `error: Call to untyped function "show" in typed context  [no-untyped-call]`.

### Code for reproduction

```python
import matplotlib.pyplot as plt
plt.show()
# Then run `mypy`.
```


### Actual outcome

error: Call to untyped function "show" in typed context  [no-untyped-call]

### Expected outcome

No error.

### Additional information

No error was reported in earlier versions.
The error appears with `matplotlib 3.8.0` and `mypy 1.5.1`.

### Operating system

Ubuntu

### Matplotlib Version

3.8.0

### Matplotlib Backend

_No response_

### Python version

_No response_

### Jupyter version

_No response_

### Installation

pip

--- a/lib/matplotlib/pyplot.py
+++ b/lib/matplotlib/pyplot.py
@@ -479,7 +479,7 @@ def draw_if_interactive(*args, **kwargs):
 
 
 # This function's signature is rewritten upon backend-load by switch_backend.
-def show(*args, **kwargs):
+def show(*args, **kwargs) -> None:
     """
     Display all open figures.
 

